### PR TITLE
feat: RootPortal Component V1

### PR DESCRIPTION
**Pull request by cwtedwards**
### Added

Added the first iteration `RootPortal` component for review and discussion. 

### Changed

- 

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/components/src/Modal/createPortal.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+export function CreatePortal({ children }: CreatePortalProps) {
+  return <Portal>{children}</Portal>;
+}
+
+interface PortalProps {
+  children: ReactNode | ReactNode[];
+}
+
+export function Portal({ children }: PortalProps) {
+  const modalRoot = document.body.appendChild(document.createElement("div"));
```
Comment (seanhealy) [inline packages/components/src/Modal/createPortal.tsx]: I think we can pull this out to the module scope. That way we don't create a new `portalRoot` for each portal.

File: packages/components/src/Modal/createPortal.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  const tempVar = document.createElement("div");
```
Comment (seanhealy) [inline packages/components/src/Modal/createPortal.tsx]: I'm trying to decide if there is any benefit to creating a temp div for each portal… I'm not thinking of any immediately I think lets simplify this part for now?

File: packages/components/src/Modal/createPortal.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import React, { ReactNode, useEffect } from "react";
+import ReactDOM from "react-dom";
+
+interface CreatePortalProps {
+  children: ReactNode | ReactNode[];
+}
+
+export function CreatePortal({ children }: CreatePortalProps) {
```
Comment (seanhealy) [inline packages/components/src/Modal/createPortal.tsx]: I don't think we need this wrapping component.

**Comment by cwtedwards** [comment]
🔱_Atlantis

File: packages/components/src/Modal/Modal.tsx [hunk 4]
```diff
@@ -88,8 +88,8 @@ export function Modal({
       )}
     </AnimatePresence>
   );
-
-  return ReactDOM.createPortal(template, document.body);
+  return <RootPortal>{template}</RootPortal>;
+  // return ReactDOM.createPortal(template, document.body);
```
Comment (seanhealy) [inline packages/components/src/Modal/Modal.tsx]:
```suggestion
```

**Comment by seanhealy** [comment]
@cwtedwards Implementation looks great! Would be good to have a little test for it and I left one tiny comment. Then I think we're good!

**Comment by cwtedwards** [comment]
@seanhealy added a snapshot test just to make sure that it will actually  render a modal as my test. Did we want this to live within the modal or do we want to break this out to be a utility component and add a section for it?

**Comment by chris-at-jobber** [comment]
@cwtedwards @seanhealy do we still need this component?

**Comment by chris-at-jobber** [comment]
Stale PR, can re-open if need for this comes up

**Pull request closed without merging**